shared/lib/trace.ts

```diff
@@ -251,3 +278,17 @@ function sentryWithIsolationScope<T>(callback: (scope: Sentry.Scope) => T): T {
 
   return actual(callback);
 }
+
+function sentrySetMeasurement(
+  key: string,
+  value: number,
+  unit: MeasurementUnit,
+) {
+  const actual = globalThis.sentry?.setMeasurement;
```

Comment:
It's odd that measurements are set using this global function, without any context of a specific span or scope. I see this is what the docs advise though.

How do they determine which span these measurements get attached do - isn't it possible for there to be multiple overlapping spans? In which case these measurements might get set during the course of one span, but then be overwritten later during the course of a later span that overlaps.

I guess that might not be that problematic for us here, since the measurements reflect global state. But if we tried to measure span-specific values, I wonder how this is supposed to work.

Reply:
Completely agree. Tags are set on a scope, and the span itself has various configuration options, but the only API I could find for measurements was indeed global.

My assumption was it's the most recent / immediate span, hence why I set it immediately after the span is created.